src/qililab/instruments/quantum_machines/quantum_machines_cluster.py

```diff
@@ -0,0 +1,354 @@
+
+                    # readout bus
+                    if "outputs" in element:
+                        element_dict["outputs"] = {
+                            key: (element["outputs"][key]["controller"], element["outputs"][key]["port"])
+                            for key in ["out1", "out2"]
+                        }
+                        element_dict["time_of_flight"] = element["time_of_flight"]
+                        element_dict["smearing"] = element["smearing"]
+                # RF with Octave
+                elif "rf_inputs" in element:
+                    element_dict["RF_inputs"] = {"port": (element["rf_inputs"]["octave"], element["rf_inputs"]["port"])}
+                    element_dict["intermediate_frequency"] = int(element["intermediate_frequency"])
+
+                    # readout bus
+                    if "rf_outputs" in element:
+                        element_dict["RF_outputs"] = {
+                            "port": (element["rf_outputs"]["octave"], element["rf_outputs"]["port"]),
+                        }
+                        element_dict["time_of_flight"] = element["time_of_flight"]
+                        element_dict["smearing"] = element["smearing"]
+
+                # other settings
+                if "digital_inputs" in element:
+                    element_dict["digitalInputs"] = {
+                        "switch": {
+                            "port": (element["digital_inputs"]["controller"], element["digital_inputs"]["port"]),
+                            "delay": element["digital_inputs"]["delay"],
+                            "buffer": element["digital_inputs"]["buffer"],
+                        }
+                    }
+                if "digital_outputs" in element:
+                    element_dict["digitalOutputs"] = {
+                        "port": (element["digital_outputs"]["controller"], element["digital_outputs"]["port"]),
+                    }
+
+                elements[bus_name] = element_dict
+
+            return elements, mixers
+
+    settings: QuantumMachinesClusterSettings
+    device: QMMDriver
+    qmm: QuantumMachinesManager
+    qm: QuantumMachine
+    config: DictQuaConfig
+    octave_config: QmOctaveConfig | None = None
+
+    def _is_connected_to_qm(self):
+        """Check if a connection to Quantum Machine is open."""
+        return hasattr(self, "qm")
+
+    @Instrument.CheckDeviceInitialized
+    def initial_setup(self):
+        """Sets initial instrument settings.
+
+        Creates an instance of the Qilililab Quantum Machines Manager, and sets the configuration dictionary.
+        """
+        if self.settings.octaves:
+            self.octave_config = QmOctaveConfig()
+            self.octave_config.set_calibration_db(os.getcwd())
+            for octave in self.settings.octaves:
+                self.octave_config.add_device_info(octave["name"], self.settings.address, octave["port"])
+        self.qmm = QuantumMachinesManager(
```

Comment:
This line is complaining locally for me saying that there is an unexpected keyword argument 'cluster_name' in the constructor call. I have checked the QuantumMachinesManager and indeed it does not have that attribute.

Reply:
Good catch, you have to update `qm_qua` to latest `1.1.6`. I will update the `requirements.txt`.

Reply:
Done